### DISCUSSION TITLE
[要望] 操作系テキストをkeybindに任せる

### DISCUSSION THREAD

**Discussion by NePonpon** [TUSB/TheUnusualSkyBlock#625] [Idea]
### アイデアのカテゴリー

インターフェイス全般

### アイデア概要

右クリックする～とかQキーを押して～とか操作系のテキストをプレイヤー環境によって可変するテキストにしたい。

### 実現方法

1. 対象となるテキストを探す。
2. with付きのテキストに修正し、keybindの部分を%1$sなどにする
3. 想定されるテキストに応じて文章を修正する

**Comment by NePonpon** [comment]
いわゆる右クリックは`key.use`で、表示上は「右ボタン」である

> 右クリック時間が長いほど
> ↓
> 右ボタン時間が長いほど

ちょっとおかしくなる

もし`key.use`をPキーに設定していたとしたら
> p時間が長いほど

になってしまう。

> %1$sを押す時間が長いほど

にするとどのキーが来ても対応できる？

**Comment by NePonpon** [comment]
マウスが3ボタン以上ついていて、サイドボタンなどを割り当てているとすると、表示が「ボタン４」になってしまう。

プレイヤーが設定したものとはいえ、いざ表示されると混乱すると思われる。

**Comment by waxsd100** [comment]
> [!NOTE]
> 要望ラベルは実装必須の**機能追加**と実装必須ではない**アイデア**が混在しているため非推奨ラベルです。
> 機能追加(実装確定)なら`機能追加`ラベル
> アイデア(要望や実装未確定)なら`アイデア`ラベル
> を追加し、`要望`ラベルは削除してください。
